setup.py

```diff
@@ -19,10 +19,10 @@
     'apache_beam>=2.31.0',
     'jax[cpu]',
     'numpy',
-    'pandas==2.0.3',
+    'pandas~=2.0.3',
     'scipy',
     'scikit-learn',
-    'xarray==2023.7.0',
+    'xarray~=2023.7.0',
```

Comment:
I don't think this is what we want. `~=2023.7.0` is equivalent to `>=2023.7.0 ==2023.7.*` which won't match any recent xarray versions (e.g. versions starting with `2024`).

Are you sure you don't mean `>=2023.7.0`? The calendar versioning has the year as the major version, and no [compatible version specifier](https://peps.python.org/pep-0440/#compatible-release) (`~=`) allows a major version bump

Reply:
good catch thanks roan i didn't take note that xarray isn't using semvar. updated to use >=